src/iuse.cpp

```diff
@@ -8083,6 +8085,324 @@ std::optional<int> iuse::capture_monster_act( Character *p, item *it, const trip
     return 0;
 }
 
+heating_requirements heating_requirements_for_weight( const units::mass &frozen,
+        const units::mass &nfrozen )
+{
+    int ammo = 2 * divide_round_up( frozen, 500_gram ) + divide_round_up( nfrozen, 500_gram );
+    int time = to_moves<int>( 20_seconds * ( frozen / 100_gram ) ) + to_moves<int>( 10_seconds *
+               ( nfrozen / 100_gram ) ) ;
+    return { ammo, time };
+}
+
+std::optional<int> iuse::heat_solid_items( Character *p, item *it, const tripoint & )
+{
+    if( p->fine_detail_vision_mod() > 4 ) {
+        p->add_msg_if_player( _( "You can't see to do that!" ) );
+        return std::nullopt;
+    }
+    if( p->cant_do_mounted() ) {
+        return std::nullopt;
+    }
+    if( !it->container_type_pockets_empty() ) {
+        p->add_msg_if_player( _( "You need an empty container to heat items." ) );
+        return std::nullopt;
+    }
+    // Check that player isn't over volume limit as this might cause it to break... this is a hack.
+    // TODO: find a better solution.
+    if( p->volume_capacity() < p->volume_carried() ) {
+        p->add_msg_if_player( _( "You're carrying too much to heat anything." ) );
+        return std::nullopt;
+    }
+    //If *it don't have container,such like COOK level 1 tools(tongs,spear), you can only heat one solid item a time(and can't be liquid), but no volume limit on each batch.
+    if( it->has_pocket_type( pocket_type::CONTAINER ) ) {
+        heat_items( p, it, false, true );
+    } else {
+        heat_single_item( p, it );
+    }
```

Comment:
These functions have a return value that you ignore. If you intend to use their return value use, otherwise don't have them return a value.
See this comment for information on what that means.
https://github.com/CleverRaven/Cataclysm-DDA/blob/cd18044375172e446488292dfe1f3a6e31934ce8/src/iuse.h#L24-L28